docs/.redirects/redirects.json

```diff
@@ -1,4 +1,5 @@
 {
+    "model-dev-guide/dtrain-introduction": "dtrain/dtrain-introduction.html",
```

Comment:
for the topic of "distributed training", the content lends itself to the model developer guide because the user is in problem-solving mode.

secondarily, it lends itself to the user who is in learning mode.

<img width="494" alt="why we show some topics in multiple places" src="https://github.com/determined-ai/determined/assets/122102764/581c9cb8-0b8a-4923-bbb2-29274ad05552">

for this reason it lives under model-dev-guide but shows up for both sections. once we split the sections into their own pages, there will be more of these "cross references"